### PR TITLE
Attack Styles: added abillity to lock widgets

### LINKED ISSUE #7903: Lock attack options
Currently on RuneLite it is possible to remove attack options which is obviously useful for pures etc, however what I am suggesting is to add an option in which you can lock the option from being selected, not removing it from the combat options tab entirely. This means that if you try to click it, it doesn't let you and potentially tells you in the chat box. This was a feature on a previous client. The purpose is it looks a lot nicer than removing options all together and is essentially nicer for clips etc.

### PR DESCRIPTION
**Pull request by Wouter8**
closes #7903

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/WarnedStyleHandleType.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package net.runelite.client.plugins.attackstyles;
+
+public enum WarnedStyleHandleType
+{
+	WARN,
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/WarnedStyleHandleType.java]: And how will you disable this? I would also add NONE here, and probably just use same enum for both auto-retalliate and warned styles

Comment (Wouter8) [reply runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/WarnedStyleHandleType.java]: You can disable everything by just disabling the style. I can't find a reason to have defence enabled, but then disable to do anything with that?

Comment (deathbeam) [reply runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/WarnedStyleHandleType.java]: Well, before this change you could just untick this, now its not possible anymore

File: runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java [hunk 2]
```diff
@@ -291,24 +344,26 @@ private void updateWarning(boolean weaponSwitch)
 				{
 					if (weaponSwitch)
 					{
-						// TODO : chat message to warn players that their weapon switch also caused an unwanted attack style change
+						sendChatMessage(WEAPON_SWITCH_MESSAGE);
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java]: Sending a notification should be a separate commit because it's solving a different issue.

File: runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/WarnedStyleHandleType.java (hunk 1 above)
Comment (Wouter8) [reply runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/WarnedStyleHandleType.java]: Untick what? I changed the 'remove warned styles' to 'how do you want to handle the styles'. You couldn't have defence enabled, but then disable the warning in another setting

File: runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java (hunk 2 above)
Comment (Wouter8) [reply runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java]: You mean a new pull request?

Comment (raiyni) [reply runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java]: no, commit.

File: runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesConfig.java [hunk 3]
```diff
@@ -98,24 +98,24 @@ default boolean warnForMagic()
...
-		keyName = "hideAutoRetaliate",
-		name = "Hide auto retaliate",
-		description = "Hide auto retaliate from the combat options tab",
+		keyName = "autoRetaliateHandler",
+		name = "How to handle auto retaliate",
+		description = "Choose how auto retaliate button will be handled",
 		position = 7
 	)
-	default boolean hideAutoRetaliate()
+	default AutoRetaliateHandleType autoRetaliateHandler()
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesConfig.java]: Maybe you should change this to boolean Warn for autoretalliate, and then adding dropdown for either locking or hiding warned stuff or do nothing

**Comment by deathbeam** [comment]
[21:41] deathbeam: NONE, NOTIFICATION, HIDE, LOCK
[21:41] deathbeam: and making autoretalliate follow same style as attack styles
[21:41] deathbeam: would make most sense imo

**Comment by Wouter8** [comment]
Alright, that seems like a good idea as well.

**Comment by deathbeam** [comment]
So then commits should be:

1. Change attack style to respect the config option for hiding warned styles, and renaming it to Warn for auto retalliate
2. Adding notification and changing the boolean to dropdown -> NONE, NOTIFICATION, HIDE
3. Adding LOCK to enum and implementing that

File: runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesConfig.java [hunk 4]
```diff
@@ -98,24 +98,24 @@ default boolean warnForMagic()
 	}
 
 	@ConfigItem(
-		keyName = "hideAutoRetaliate",
-		name = "Hide auto retaliate",
-		description = "Hide auto retaliate from the combat options tab",
+		keyName = "autoRetaliateHandler",
+		name = "How to handle auto retaliate",
```
Comment (Hydrox6) [inline runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesConfig.java]: These names are just a bit too long to fit in the config panel, and could be made less wordy

**Comment by Wouter8** [comment]
Recommitted with wanted changes

**Comment by Hydrox6** [comment]
Looks like you'll need to update the test for this plugin. Check the travis log

File: runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesConfig.java [hunk 5]
```diff
@@ -98,24 +98,24 @@ default boolean warnForMagic()
...
-		description = "Hide auto retaliate from the combat options tab",
+		keyName = "warnForAutoRetaliate",
+		name = "Warn for auto retaliate",
+		description = "Show warning when auto retaliate is selected",
 		position = 7
 	)
-	default boolean hideAutoRetaliate()
+	default boolean warnForAutoRetaliate()
 	{
 		return false;
```
Comment (raiyni) [inline runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesConfig.java]: why is auto retaliates format specifically warn?

Comment (Wouter8) [reply runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesConfig.java]: See thread above. Changed so that auto retaliate is handled same as other buttons. There's 1 new setting in which you can specify if you want to: do nothing, be warned, hide buttons or lock buttons. 

Now that you mention it: I might need to add a chatmessage when the option is set to 'warn' and auto retaliate switches.

**Comment by Wouter8** [comment]
@deathbeam fixed your requested changes

File: runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java [hunk 6]
```diff
@@ -168,14 +182,22 @@ private void processWidgets()
...
+			for (WidgetInfo widgetKey : widgetsToHideOrLock.row(equippedWeaponType).keySet())
 			{
-				hideWidget(client.getWidget(widgetKey), widgetsToHide.get(equippedWeaponType, widgetKey));
+				if (config.warnedStyleHandler() == HandleType.HIDE)
+				{
+					hideWidget(client.getWidget(widgetKey), widgetsToHideOrLock.get(equippedWeaponType, widgetKey));
+				}
+				else
+				{
+					// Always show widget when not on 'REMOVE' option
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java]:
```suggestion
					// Always show widget when not on 'HIDE' option
```

File: runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java [hunk 7]
```diff
@@ -191,6 +213,51 @@ public void onGameStateChanged(GameStateChanged event)
 		}
 	}
 
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		// Warn for auto-retaliate?
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/attackstyles/AttackStylesPlugin.java]: Is this a TODO comment? It should either be implemented if you meant to implement it, or removed otherwise.

**Comment by Wouter8** [comment]
After almost 4 months of inactivity I'm not interested in making changes to this pull request anymore.
